docs/reference/modules/module.md

```diff
@@ -0,0 +1,103 @@
+
+Returns rendered HTML for a template with the data provided. You must pass `req`, a request object, as the first argument. The `template` argument should be the name of a template file in the module's `views` directory.
+
+All properties of `data` can be used in Nunjucks templates as properties of the `data` object. This argument may be omitted to include no additional data.
+
+### `async renderString(req, string, data)`
+
+Returns rendered HTML for a Nunjucks-style string with the data provided. You must pass `req`, a request object, as the first argument. The `string` argument will be used as the template itself for rendering (not the template filename).
+
+All properties of `data` can be used in Nunjucks templates as properties of the `data` object. This argument may be omitted to include no additional data.
+
+### `async send(req, template, data)`
+
+The `send()` method renders a template with data as with the [`render()`](#async-render-req-template-data), then sends the rendered HTML as a response to the request (`req`).
+
+### `async sendPage(req, template, data)`
+
+Similarly to the [`send()`](#async-send-req-template-data) method renders a template and sends the rendered HTML as a response to the request (`req`). Where `send()` is used to render general template files, `sendPage()` is specifically used to render and send *full pages* for Apostrophe projects.
+
+Templates rendered and sent with this method have full access to all template [data properties](/guide/template-data.md) appropriate for the related module. The template is also wrapped with the proper layout file (`'@apostrophecms/template:outerLayout.html'` by default), including the full `head` tag.
+
+This method triggers the `@apostrophecms/page` module to emit a [`beforeSend` event](/reference/server-events.md#beforesend).
+
+### `enableBrowserData(scene)`
+
+Call this method from a module's [initialization function](/reference/module-api/module-overview.md##initialization-function) if the module implements the [`getBrowserData()` method](#getbrowserdata-req). **This is only necessary for modules that do not already do this** (all doc type and widget type modules already do this). The data returned by `getBrowserData(req)` will then be available on `apos.modules['the-module-name']` in the browser.
+
+The `scene` argument is only needed if making the data available while logged-out. By default browser data is pushed only for the `apos` scene (the logged-in context), so the anonymous site visitor experience will include the extra data (except on the /login page and other pages that opt into the `apos` scene). If `scene` is set to `public` then the data is available to all visitors.
+
+### `getBrowserData(req)`
+
+The `getBrowserData()` method can be extended to adjust or add data that the module makes available in the browser. It should return an object to make the data available in browsers (as controlled by [the `enableBrowserData()` method](#enablebrowserdata-scene)).
```

Comment:
Feel like we need a good mnemonic indicator for "this is a method you implement, not a method you call."

Reply:
Yeah, I realized a bit into this one how that's a real distinction here. This would be a good case for a code recipe article.